src/Bicep.Core/Analyzers/Linter/Rules/UseSafeAccessRule.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.CodeAction;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.Syntax;
+using Bicep.Core.Syntax.Comparers;
+using Bicep.Core.Syntax.Visitors;
+
+namespace Bicep.Core.Analyzers.Linter.Rules;
+
+public sealed class UseSafeAccessRule : LinterRuleBase
+{
+    public new const string Code = "use-safe-access";
+
+    public UseSafeAccessRule() : base(
+        code: Code,
+        description: CoreResources.UseSafeAccessRule_Description,
+        LinterRuleCategory.BestPractice,
+        docUri: new Uri($"https://aka.ms/bicep/linter/{Code}"))
+    { }
+
+    public override IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model, DiagnosticLevel diagnosticLevel)
+    {
+        foreach (var ternary in SyntaxAggregator.AggregateByType<TernaryOperationSyntax>(model.Root.Syntax))
+        {
+            if (SemanticModelHelper.TryGetNamedFunction(model, SystemNamespaceType.BuiltInName, "contains", ternary.ConditionExpression) is not {} functionCall ||
+                functionCall.Arguments.Length != 2 ||
+                functionCall.Arguments[1].Expression is not StringSyntax containsString ||
+                containsString.TryGetLiteralValue() is not {} propertyName)
```

Comment:
Would it be worth adding a TODO comment about supporting cases where the second argument is an expression? Thinking of examples like:

```bicep
param foo object
var propertyName = 'bar'
var foobar = contains(foo, propertyName) ? foo[propertyName] : 'default' // -> foo[?propertyName] ?? 'default'
```

This would be more complicated but is probably worth supporting in the long run.

Reply:
Good call, I added this